crates/provider-wit-bindgen/src/lib.rs

```diff
@@ -41,187 +55,926 @@ const EXPORTS_MODULE_NAME: &str = "exports";
+                if let Some(name) = &type_def.name {
+                    name.as_str()
+                } else {
+                    panic!(
+                        r#"
+Un-named type detected on argument while parsing interface [{}].
+Please ensure that types for exported interface functions (normally a single record type representing a message to be sent on the lattice) are specified.
+"#,
+                        iface.name.clone().unwrap_or("<unknown>".into()),
+                    )
+                }
+            }
+        };
+
+        let fn_name = Ident::new(iface_fn_name.to_snake_case().as_str(), Span::call_site());
+
+        let lattice_method = LitStr::new(
+            format!("Message.{}", iface_fn_name.to_upper_camel_case()).as_str(),
+            Span::call_site(),
+        );
+
+        let arg_name_ident = Ident::new(arg_name, Span::call_site());
+        let arg_type_ident = Ident::new(rust_type, Span::call_site());
+
+        let contract_ident = LitStr::new(&cfg.contract, Span::call_site());
+
+        Ok(quote::quote!(
+            async fn #fn_name(
+                &self,
+                #arg_name_ident: #arg_type_ident,
+            ) -> Result<(), ::wasmcloud_provider_sdk::error::ProviderInvocationError> {
+                let connection = ::wasmcloud_provider_sdk::provider_main::get_connection();
+                let client = connection.get_rpc_client();
+                let response = client
+                    .send(
+                        ::wasmcloud_core::WasmCloudEntity {
+                            public_key: self.ld.provider_id.clone(),
+                            link_name: self.ld.link_name.clone(),
+                            contract_id: #contract_ident.to_string(),
+                        },
+                        ::wasmcloud_core::WasmCloudEntity {
+                            public_key: self.ld.actor_id.clone(),
+                            ..Default::default()
+                        },
+                        #lattice_method,
+                        ::wasmcloud_provider_sdk::serialize(&#arg_name_ident)?,
+                    )
+                    .await?;
+
+                if let Some(err) = response.error {
+                    Err(::wasmcloud_provider_sdk::error::ProviderInvocationError::Provider(err.to_string()))
+                } else {
+                    Ok(())
+                }
+            }
+        ))
+    }
+
+    /// Translate an exported WIT function via bundled arguments
+    fn translate_export_fn_via_bundled_args(
+        iface: &wit_parser::Interface,
+        _iface_fn_name: &str,
+        _iface_fn: &wit_parser::Function,
+        _cfg: &ProviderBindgenConfig,
+    ) -> anyhow::Result<TokenStream> {
+        bail!(
+            r#"[warn] Functions on exported interfaces must only accept one argument, normally a record type that can be sent out on the lattice. Please modify the corresponding WIT file that contains interface [{}]"#,
```

Comment:
Help me understand why this limitation exists?

Reply:
Ah it's just not implemented yet -- this was added back @ v1 of the bindgen working on the early `messaging` contract example -- this limitation can be removed in the future